### ISSUE TITLE
[Question] torch's `.backward()` equivalent

### ISSUE THREAD

**Issue by a1eaiactaest** [ml-explore/mlx#605]
How would thix PyTorch code look in MLX?
```python3
loss = torch.mean(p_losses + v_losses)
loss.backward()
```
I am aware of the fact that `backward` isn't needed according to the [docs](https://ml-explore.github.io/mlx/build/html/usage/function_transforms.html?highlight=backward#automatic-differentiation), but am not sure how to reimplement it. Any help and advice is appreciated.

**Comment by awni** [comment]
Well in your case what are you trying to get the gradient of?

As a simple example:

```python
def fun(x, y):
    return mx.mean(x+ y)

loss, (dfdp_losses, dfdv_losses) = mx.value_and_grad(fun)(p_losses, v_losses)
```
